src/probnum/linalg/solvers/_probabilistic_linear_solver.py

```diff
@@ -156,13 +158,17 @@ def solve_iterator(
 
             yield solver_state
 
+            # Check stopping criterion
+            if self.stopping_criterion(solver_state=solver_state):
+                break
```

Comment:
It might be useful to check the stopping criterion before the `yield`, write the result into the `state` as a `bool`, then `yield` and then `break` depending on the value of that `bool`. This way the user knows whether the solver is about to stop. However, from my POV, this is out of scope for this PR